src/components/common/Button/Button.tsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import * as styled from './Button.styles';
+
+type ButtonProps = Omit<
+  React.DetailedHTMLProps<React.ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>,
+  'ref'
+> &
+  React.PropsWithChildren<{
+    variant?: 'primary' | 'outlined';
+  }>;
```

Comment:
```suggestion
type ButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement> &
  React.PropsWithChildren<{ variant?: 'primary' | 'outlined' }>;
```

이렇게 하는 거랑 결과가 다른가용? (궁금)

Reply:
결과는 같네요! 😅 해당 케이스에서 간소화 하는게 좋아보여 리팩토링 진행했습니다! 🫡

```ts
type ButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement> &
  React.PropsWithChildren<{ variant?: 'primary' | 'outlined' }>;
```